assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -31,45 +31,67 @@ import Context from './context';
 function MediaProvider({ children }) {
   const { state, actions } = useMediaReducer();
   const { uploadVideoFrame } = useUploadVideoFrame();
-  const { mediaType, searchTerm } = state;
+  const { page, mediaType, searchTerm } = state;
   const {
     fetchMediaStart,
     fetchMediaSuccess,
     fetchMediaError,
     resetFilters,
     setMediaType,
     setSearchTerm,
+    setPage,
   } = actions;
 
   const {
     actions: { getMedia },
   } = useAPI();
 
-  const fetchMedia = useCallback(() => {
-    fetchMediaStart();
-    getMedia({ mediaType, searchTerm })
-      .then((media) => {
-        fetchMediaSuccess({ media, mediaType, searchTerm });
-      })
-      .catch(fetchMediaError);
-  }, [
-    fetchMediaError,
-    fetchMediaStart,
-    fetchMediaSuccess,
-    getMedia,
-    mediaType,
-    searchTerm,
-  ]);
+  const fetchMedia = useCallback(
+    ({ page: p = 1 } = {}) => {
+      fetchMediaStart({ page: p });
+      getMedia({ mediaType, searchTerm, page: p })
+        .then(({ data, headers }) => {
+          const totalPages = parseInt(headers.get('X-WP-TotalPages'));
+          fetchMediaSuccess({
+            media: data,
+            mediaType,
+            searchTerm,
+            page: p,
+            totalPages,
+          });
+        })
+        .catch(fetchMediaError);
+    },
+    [
+      fetchMediaError,
+      fetchMediaStart,
+      fetchMediaSuccess,
+      getMedia,
+      mediaType,
+      searchTerm,
+    ]
+  );
+
+  const resetAfterUpload = useCallback(() => {
```

Comment:
Perhaps rename this `resetWithFetch` to make it more general and applicable in the future for other cases, too?

Reply:
Done https://github.com/google/web-stories-wp/pull/562/commits/1ffa6d3b720112520e9bdd111c31d9f5907fd3dd